pythonforandroid/distribution.py

```diff
@@ -19,14 +23,21 @@ class Distribution(object):
 
     name = None  # A name identifying the dist. May not be None.
     needs_build = False  # Whether the dist needs compiling
+    needs_clean_build = False  # Whether the build needs a full clean
     url = None
     dist_dir = None  # Where the dist dir ultimately is. Should not be None.
     ndk_api = None
+    android_api = None
+    p4a_version = None
 
     archs = []
     '''The arch targets that the dist is built for.'''
 
-    recipes = []
+    recipes = {}
+    '''A dictionary which holds recipes information. Each key is the name of
+    the recipe, and the value is recipe's version'''
+
+    recipes_to_rebuild = set()
```

Comment:
Minor: why not being consistent and use the literal `{}`?

Reply:
I don't think that python have a literal for an empty set...`{}` this will be interpreted as  a dict, unless we put some values in the set creation...I'm right doesn't?

Reply:
You're completely right! And I should get more sleep :sweat_smile: